hardware/opas/TOPAS/TOPAS.py

```diff
@@ -176,27 +104,31 @@ def _load_curve(self, interaction):
             del curve_paths_copy['Poynting']
         crv_paths = [m.read() for m in curve_paths_copy.values()]
         all_curves = attune.TopasCurve.read_all(crv_paths)
+        for curve in all_curves.values():
+            for dependent in curve.dependent_names:
+                if dependent not in self.motor_names:
+                    try:
+                        curve.rename_dependent(dependent, self.motor_names[int(dependent)])
+                    except:
+                        pass
         self.interaction_string_combo.set_allowed_values(list(all_curves.keys()))
         self.curve = all_curves[interaction]
         return self.curve
        
     def _set_motors(self, motor_destinations):
         for motor_name, destination in motor_destinations.items():
-            motor_index = self._get_motor_index(motor_name)
-            error, destination_steps = self.api.convert_position_to_steps(motor_index, destination)
-            self.api.start_motor_motion(motor_index, destination_steps)
+            destination = float(destination)
+            self.api.set_motor_position(motor_name, destination)
 
     def _update_api(self, interaction):
```

Comment:
This function is less useful if WinTOPAS3 is not being used.

Technically, wintopas3 still can be used, I have not changed curve handling
WinTopas4 does not provide an obvious programmatic way to update curves, would have to rewrite.

I believe that the curves are now JSON specified (In a file called something like "Optical System"). We do not have a parser/generator though